bin/automatic_review.py

```diff
@@ -0,0 +1,232 @@
+
+                if review == 'pass':
+                    patch_review['reviewed'] = 'pass'
+                    rest_communication.patch_entries('run_elements', payload=patch_review, update_lists=None, where={"run_id":self.run_name, "lane": lane})
+                elif review == 'fail':
+                    review_comments = ('lane ' + str(lane) + ' failed due to ' + ' '.join(run_review[lane][1]))
+                    patch_review['reviewed'] = 'fail'
+                    patch_comments[ELEMENT_REVIEW_COMMENTS] = review_comments
+                    rest_communication.patch_entries('run_elements', payload=patch_review, update_lists=None, where={"run_id":self.run_name, "lane": (lane)})
+                    rest_communication.patch_entries('run_elements', payload=patch_comments, update_lists=None, where={"run_id":self.run_name, "lane": (lane)})
+
+class AutomaticSampleReview(SampleReview):
+    def __init__(self, sample_name, sample):
+        self.sample = sample
+        self.sample_name = sample_name
+        super().__init__(sample)
+
+    def patch_entry(self):
+        sample_review = self.sample_review()
+        review = ''
+        if sample_review:
+            if sample_review == 'pass':
+                review = sample_review
+            else:
+                review = (list(sample_review.values())[0])[0]
+            print(review)
+            patch_review = {}
+            patch_comments = {}
+
+            if review == 'pass':
+                patch_review['reviewed'] = 'pass'
+                rest_communication.patch_entries('samples', payload=patch_review, update_lists=None, where={"sample_id": self.sample_name})
+
+            else:
+                patch_review['reviewed'] = 'fail'
+                patch_comments[ELEMENT_REVIEW_COMMENTS] = (('sample ' + self.sample_name + ' failed due to ' + (' '.join((list(sample_review.values())[0])[1]))))
+                rest_communication.patch_entries('samples', payload=patch_review, update_lists=None, where={"sample_id": self.sample_name})
+                rest_communication.patch_entries('samples', payload=patch_comments, update_lists=None, where={"sample_id": self.sample_name})
+
+def get_reviewable_runs():
+    runs = rest_communication.get_documents('aggregate/all_runs', depaginate=True, match={"proc_status":"finished","review_statuses":"not%20reviewed"})
+    for run in runs:
+        run_id = run.get('run_id')
+        run_elements_by_lane = rest_communication.get_documents('aggregate/run_elements_by_lane', match={"run_id":run_id})
+        r = AutomaticRunReview(run_id, run_elements_by_lane)
+        run_review = r.patch_entry()
+
+def get_reviewable_samples():
+    samples = rest_communication.get_documents('aggregate/samples', depaginate=True,  match={"proc_status":"finished","reviewed":"not%20reviewed"})
+    for sp in samples:
+        sample_id = sp.get('sample_id')
+        sample = rest_communication.get_documents('aggregate/samples', match={"sample_id":sample_id})
```

Comment:
If you use `get_document` (rather than `get_documents`), there will be no danger of looping over an empty list in `SampleReview.sample_metrics` and failing silently